packages/next/lib/file-exists.ts

```diff
@@ -0,0 +1,13 @@
+import fs from 'fs'
+import { promisify } from 'util'
+
+const access = promisify(fs.access)
+
+export async function fileExists(fileName: string): Promise<boolean> {
+  try {
+    await access(fileName, fs.constants.F_OK)
+    return true
+  } catch (_) {
```

Comment:
This should probably check for the specific code, I think it's `ENOENT` from the top of my mind. Otherwise it catches all errors.